blogposts/2020/optimizing-a-code-intelligence-indexer.md

```diff
@@ -0,0 +1,179 @@
+---
+title: "Optimizing a code intelligence indexer"
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+publishDate: 2020-08-17T08:00-07:00
+tags: [blog]
+slug: optimizing-a-code-intel-indexer
+heroImage: https://sourcegraphstatic.com/blog/lsif-go/lsif-go-improvements.png
+published: true
+---
+
+On the [code intelligence team](https://about.sourcegraph.com/handbook/engineering/code-intelligence) at Sourcegraph, we want you to always have precise and up-to-date hovers, definitions, and references when you're browsing and reviewing unfamiliar code, even on very large monorepos.
+
+![Cross-repository jump to definition](https://sourcegraphstatic.com/precise-xrepo-j2d.gif)
+
+We call our key (internal) metric **time to intelligence** (TTI), which covers the time it takes to:
+
+1. Push a new commit to a branch on your repository
+1. Trigger a CI job to index the code of that commit
+1. Upload the resulting index to your Sourcegraph instance
+1. Process the uploaded index
+1. Use the processed index to provide up-to-date code intelligence
+
+Sourcegraph is used by companies all over the repository number and size spectrum. Some companies employ a microservice approach. For us, that's easy mode. Small repositories are easy and generally fast _enough_ to index even with an indexer that becomes slow as code size grows. Other companies employ a monorepo approach. This is where the challenge lies for us - but we've made great strides with the recent [1.0 release](https://github.com/sourcegraph/lsif-go/releases/tag/v1.0.0) of [lsif-go](https://github.com/sourcegraph/lsif-go).
+
+Now, how well do we index Go at scale?
```

Comment:
```suggestion
Now let's take a look at how well we are indexing Go at scale using the TTI metric as our indicator:
```

Reply:
The problem with this suggestion is that we're not measuring all of TTI here, only the indexing portion.

Reply:
What role does TTI play in this optimization then?

Reply:
We're optimizing it, but the measurements here are only for the indexing portion.